puente-etl/utils/constants.py

```diff
@@ -0,0 +1,104 @@
+class ColumnOrder:
+    # Enforce column order for relational table outputs
+    FORM_SPECIFICATIONS = [
+        'custom_form_id',
+        'custom_form_organizations',
+        'custom_form_class',
+        'custom_form_type_of_form',
+        'custom_form_name',
+        'custom_form_description',
+        'custom_form_header',
+        #
+        # Excluding these for now for readability:
+        #
+        # 'custom_form_updated_at',
+        # 'custom_form_created_at',
+        # 'custom_form_custom_form',
+        # 'custom_form_workflows',
+        # 'custom_form_active',
+        'question_id',
+        'question_label',
+        'question_text',
+        'question_field_type',
+        'question_formik_key',
+        'question_value',
+        #
+        # Excluding these for now for readability:
+        #
+        # 'question_active',
+        # 'question_number_questions_to_repeat',
+        # 'question_side_label',
+        # 'question_validation',
+        'answer_id',
+        'answer_label',
+        'answer_value',
+        'answer_text',
+        'answer_text_key',
+        'answer_text_question'
+    ]
+
+    FORM_RESULTS = [
+        'form_result_id',
+        'form_result_surveying_organization',
+        'form_result_surveying_user',
+        'custom_form_id',
+        'form_result_response_id',
+        'question_title',
+        'question_answer',
+        'question_type',
+        #
+        # Excluding these for now for readability:
+        #
+        # 'custom_form_name',
```

Comment:
need `custom_form_name` but could probably slugify this or `camelCase`, whichever you prefer

Reply:
The constants file mostly shows the post-processing column names. I converted to snake case out of preference. Column names can get pretty long, also when working with data you often have to use the suffix "Id" which I do not like when used in camelCase. 

I also gave them some bespoke names to help reduce mental load. If you want to maintain the camelCase columns instead, as they are stored in MongoDB, you can remove the usage of the [to_snake_case helper function](https://github.com/bcrant/serverless/blob/5d0df833bcc2faef9f232e61081cb225deab5262/puente-etl/utils/helpers.py#L14). You'd also have to refactor some string manipulation through the `puente-etl` dir. Back4App stores its default tables (Users, Sessions, Roles) with double underscores that I removed.

Reply:
In this instance, the Form Specifications are joined to Form Results so `custom_form_name` was excluded to prevent overlapping columns. Pandas creates `custom_form_name_x` and `custom_form_name_y` when this happens –– I've never liked that as the default behavior, so I try to eliminate upstream. I will leave this one for now.